api/models/Tag.js

```diff
@@ -146,23 +133,43 @@ module.exports = bookshelf.Model.extend({
   }
 
 }, {
+  addToCommunity: ({ community_id, tag_id, user_id, description, is_default }, opts) =>
+  CommunityTag.where({community_id, tag_id}).fetch(opts)
+  .tap(comTag => comTag ||
+    CommunityTag.create({community_id, tag_id, user_id, description, is_default}, opts)
+    .catch(() => {}))
+    // the catch above is for the case where another user just created the
+    // CommunityTag (race condition): the save fails, but we don't care about
+    // the result
+  .then(comTag => comTag && comTag.save({updated_at: new Date()}))
+  .then(() => user_id &&
+    TagFollow.where({community_id, tag_id, user_id}).fetch(opts)
+    .then(follow => follow ||
+      TagFollow.create({community_id, tag_id, user_id}, opts))),
+
   isValidTag: function (text) {
-    return !!text.match(/^[A-Za-z][\w-]+$/)
+    return !validateTopicName(text)
+  },
+
+  validate: function (text) {
+    return validateTopicName(text)
   },
 
   tagsInText,
 
-  find: function (id, options) {
-    if (!id) return Promise.resolve(null)
-    if (isNaN(Number(id))) {
-      return Tag.query(qb => qb.whereRaw('lower(name) = lower(?)', id))
-      .fetch(options)
+  find: function ({ id, name }, options) {
+    if (id) {
+      return Tag.where({ id }).fetch(options)
+    }
+    if (name) {
+      return Tag.query(qb => qb.whereRaw('lower(name) = lower(?)', name))
+        .fetch(options)
     }
-    return Tag.where({id: id}).fetch(options)
+    return Promise.resolve(null)
```

Comment:
sure you wanna return resolve with null?  any repercussions?

Reply:
That was the existing behaviour, so I didn't wanna mess with it. In this case I think it will simply go away when Robbie dumps the table :smile: